newletter.md

```diff
@@ -0,0 +1,104 @@
+ - We rely on [ElasticSearch](https://www.elastic.co/fr/products/elasticsearch) for searching e-mails
+ - We authenticate our users with LDAP
+ - We use [Guice](https://github.com/google/guice/wiki/GettingStarted) for bringing all pieces together.
+ - We plan to bring soon Sieve and ManageSieve part of this deployment as many peaple request filtering.
+
+Let me share a few numbers with you :
+
+ - We handle about 100 users
+ - We have so far a number of 1.500.000 e-mails, that we imported using [IMAP-sync](https://imapsync.lamiral.info/) scripts
+ - We receive around 15.000 incoming e-mails a day
+ - We execute 42.500 IMAP commands a day
+ - We answer 20.000 JMAP requests a day
+
+This deployment help us detecting bugs, and performance issues. We follow the current state of master branch, that we update at midnight 
+with the built image uploaded on [dockerhub](https://hub.docker.com/r/linagora/james-ldap-project/).
+
+## Tracking performance
+
+Everybody wants to read their e-mails fast. Thus the team made performance tracking a priority. And we developed the tools to follow performance.
+
+![Result of 1.000 thunderbird like users](gatling1.png)
+
+We started implementing [Gatling](http://gatling.io/) load testing for the James server :
+
+  - First with a [Gatling JMAP](https://github.com/linagora/james-gatling) implementation.
+  - Then we added a naive SMTP implementation
+  - And finally contributed a [IMAP DSL for Gatling](https://github.com/linagora/gatling-imap)
+
+![Result graph of 1.000 thunderbird like users](gatling2.png)
+
+Running these load tests we :
+
+  - Succeeded to support 1.000 thunderbird like users
+  - However, it turned out we had problems with 10.000 users.
+
+Additionally, we added metrics a bit everywhere in James using the brand new metrics API. We collect and export everything in [ElasticSearch](https://github.com/elastic/elasticsearch-metrics-reporter-java) using [Dropwizard metrics](http://metrics.dropwizard.io/3.2.1/). Then we graph it all using [Grafana](http://grafana.com/). This allow us to collect all statistic and percentiles. We track so far :
+
+  - Protocols detailed time execution (and count)
+  - Percentile of mailet and matcher executions
+  - Enqueue and Dequeue time
+  
+![James metrics and percentile for the JMAP protocol](grafana.png)
+
+All these solutions allow us to identify the components that needs improvement. For instance receiving too much incoming e-mails overload James with very heavy garbage collection. We then plan to move our mail queue to RabbitMQ, to parse e-mails only once on top of the mail pipeline, to reorganize a bit our configuration. A Camel upgrade (impossible in java-6) might also help.
+ 
+ ## MessageId refactoring
+ 
+ We succeeded to finish and merge the MessageId refactoring. This huge task allow us to address messages by their ID, outside of mailbox context.
+ 
+ This is required for JMAP protocol implementation.
```

Comment:
For the record that's not exactly required. You can have an implementation where each message belongs only to one folder, but then you cannot handle folders as labels.